### PR TITLE
test(autoware_traffic_light_arbiter): add node test

### PR DESCRIPTION
**Pull request by MasatoSaeki**
## Description
This PR add node alive test on default parameter.

This test show following three behavior.
- When this package node subscribes only perception msg, publish that message as it is.
- When this package node subscribes only external msg, publish that message as it is.
- when this package node subscribes both of perception and external msg, publish message which has higher confidence.

## Related links
None.

## How was this PR tested?
run below command
1. build for test
```
colcon build --merge-install --symlink-install --packages-up-to autoware_traffic_light_arbiter --cmake-args -DCMAKE_BUILD_TYPE=Debug
```
2. run test
```
colcon test --merge-install --packages-select autoware_traffic_light_arbiter --event-handlers console_cohesion+
```
You can see detail in `log/latest_test/autoware_traffic_light_arbiter/stdout.log`

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_traffic_light_arbiter/test/test_node.cpp [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+}
+
+bool isMsgEqual(const TrafficSignalArray & input_msg, const TrafficSignalArray & gt_msg)
+{
+  // check number of groups
+  if (input_msg.traffic_light_groups.size() != gt_msg.traffic_light_groups.size()) {
+    return false;
+  }
+
+  for (size_t group_idx = 0; group_idx < input_msg.traffic_light_groups.size(); ++group_idx) {
```
Comment (knzo25) [inline perception/autoware_traffic_light_arbiter/test/test_node.cpp:119]:
```suggestion
  for (std::size_t group_idx = 0; group_idx < input_msg.traffic_light_groups.size(); ++group_idx) {
```

File: perception/autoware_traffic_light_arbiter/test/test_node.cpp [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    }
+
+    // check number of elements
+    if (input_traffic_light_group.elements.size() != gt_traffic_light_group.elements.size()) {
+      std::cout << input_traffic_light_group.elements.size() << std::endl;
+      std::cout << gt_traffic_light_group.elements.size() << std::endl;
+      return false;
+    }
+
+    for (size_t element_idx = 0; element_idx < input_traffic_light_group.elements.size();
```
Comment (knzo25) [inline perception/autoware_traffic_light_arbiter/test/test_node.cpp:137]:
```suggestion
    for (std::size_t element_idx = 0; element_idx < input_traffic_light_group.elements.size();
```

File: perception/autoware_traffic_light_arbiter/test/test_node.cpp (hunk 1 above)
Comment (MasatoSaeki) [reply perception/autoware_traffic_light_arbiter/test/test_node.cpp]: Thank you for your review. I fixed in https://github.com/autowarefoundation/autoware.universe/pull/8747/commits/58c928b160251fd8a374964eeb2ee3eeba9002ac

File: perception/autoware_traffic_light_arbiter/test/test_node.cpp (hunk 2 above)
Comment (MasatoSaeki) [reply perception/autoware_traffic_light_arbiter/test/test_node.cpp]: Thank you for your review. I fixed in https://github.com/autowarefoundation/autoware.universe/pull/8747/commits/58c928b160251fd8a374964eeb2ee3eeba9002ac

**Review by knzo25 — APPROVED** [review]
LGTM!
(Btw, you can just accept the suggestions with the commit suggestion button instead of sending your own commit to save time)

**Comment by go-sakayori** [comment]
@MasatoSaeki 
There seems to be an error in the build-and-test action. Can you fix it?
image

**Comment by knzo25** [comment]
Maybe adding a `#include <autoware_lanelet2_extension/utility/message_conversion.hpp>` works?

**Comment by go-sakayori** [comment]
@knzo25 As you've mentioned it did work, so I'll create a PR.

**Comment by knzo25** [comment]
@MasatoSaeki 
Sorry I did not realize it, but one coding guideline is "include what you use" :pray: 
Since some headers change depending how you compile, this sometimes happens :cry:

**Comment by MasatoSaeki** [comment]
@go-sakayori @knzo25 
I am sorry and thank you for prompt response :bow:
> Sorry I did not realize it, but one coding guideline is "include what you use" 🙏
Since some headers change depending how you compile, this sometimes happens 😢

I see. Thank for your guidance.

**Pull request merged**